scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java

```diff
@@ -385,24 +385,19 @@ public static Job parseJSONInfo(JSONObject jsonJobInfo) {
 
     private static Map<String, String> extractGenericInformation(JSONValue jsonGenericInformation) {
         if (jsonGenericInformation != null) {
-            JSONArray genericInformationArray = jsonGenericInformation.isArray();
-            if (genericInformationArray != null) {
-                Map<String, String> genericInformation = new HashMap<>(genericInformationArray.size());
-                for (int i = 0; i < genericInformationArray.size(); i++) {
-                    JSONObject genericInformationObject = genericInformationArray.get(i).isObject();
-                    genericInformation.put(getJsonStringValue(genericInformationObject, "key"),
-                                           getJsonStringValue(genericInformationObject, "value"));
+            JSONObject genericInformationObject = jsonGenericInformation.isObject();
+            if (genericInformationObject != null) {
+                Set<String> genericInformationKeySet = genericInformationObject.keySet();
+                Map<String, String> genericInformation = new HashMap<>(genericInformationKeySet.size());
+                for (String key : genericInformationKeySet) {
```

Comment:
it would be better to loop with Map.Entry instead of getting keySet first and loop on the key set, the code will be simpler.